addons/common/XEH_preInit.sqf

```diff
@@ -291,9 +291,22 @@ GVAR(syncedEvents) = HASH_CREATE;
 //Debug
 ACE_COUNTERS = [];
 
-// Load settings
+// Wait for server settings to arrive
+GVAR(SettingsInitialized) = false;
+["ServerSettingsReceived", {
+    // Load user settings from profile
+    if (hasInterface) then {
+        call FUNC(loadSettingsFromProfile);
+        call FUNC(loadSettingsLocalizedText);
+    };
+    GVAR(SettingsInitialized) = true;
+}] call FUNC(addEventhandler);
+
+// Load settings on the server and broadcast them
 if (isServer) then {
     call FUNC(loadSettingsOnServer);
+    // Raise a local event for other modules to listen too
+    ["SettingsInitialized", []] call FUNC(localEvent);
```

Comment:
This is not true in case the server also is a client. It still has to set the client settings to correct values.